build/scripts/process_build.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// @ts-nocheck <-- This is needed as this file is basically a JavaScript script
+//                 but with some TypeScript niceness baked in
+const fs = require('fs')
+const execSync = require('child_process').execSync
+
+function prebuild() {
```

Comment:
What is this doing? Add comment that this "Combines license and notice information into the bin.License or whatever so that we don't have to maintain the bin.NOTICE manually. Assuming I've guessed correctly why we need this. 

Why are all these calls "Sync" variants? What's happening that requires that?

Reply:
So what this script is doing is replacing the OS specific commands to move `bin.LICENSE` and `bin.NOTICE` to be `LICENSE` and `NOTICE` before packaging the VSIX, then reverting them back once the VSIX is done.

So the sync variants just ensure that nothing else is going to happen before this item completes its tasks.

Reply:
I guess I'd like to see comments about why we need this contortion? I.e., what are we doing here? Why do we need LICENCE and NOTICE in the VSIX (is that where vscode expects to find? is this an Apache requirement that packaged releases like the vsix have these  and why the bin.LICENSE, and bin.NOTICE naming convention?  What does that achieve for us?

This is probably also done in Daffodil libraries, I just don't know that any of these requirements are written down anywhere which is my frustration. The scripting always seems fragile and unmotivated to me, as it is addressing a bunch of unstated requirements. 

I'll think about creating a JIRA ticket to document these build-an-artifact requirements for LICENSE/NOTICE and how we make that work in a modular fashion, etc. 

No action required for this comment. Just thinking aloud.